src/api/app/views/webui/kiwi/repositories/edit.html.haml

```diff
@@ -0,0 +1,32 @@
+= render partial: 'webui/kiwi/tabs'
+
+%h2
+  = @image.name
+%h3 Repositories
+- if @repositories.empty?
+  %p There are no repositories.
+- else
+  %div
+    = form_for @image, as: :image, url: kiwi_image_repositories_update_path(@image), method: :post, html: { id: 'update_form' } do |f|
+      = f.fields_for :repositories do |repository_fields|
+        = render 'repository_fields', f: repository_fields
+      = button_tag('Save', id: 'update_form_save', type: 'button', disabled: true)
+
+:javascript
+  function save_repositories() {
+    $.ajax({ url: "#{url_for(controller: 'kiwi/images', action: :show, id: @image)}",
+      dataType: 'json',
+      success: function(json) {
+        var is_outdated = json['is_outdated'];
+        if (is_outdated && !confirm("This repositories have been modified meanwhile you were editing.\nDo you want to apply the changes anyway?"))
+          return;
+        $('#update_form').submit();
+      }
+    });
+  }
+
+  $('#update_form_save').click(save_repositories);
```

Comment:
Neat but very shaky....

The only thing that keeps you from accidentally updating the kiwi file is this javascript. Which can go wrong for a million reasons.

I would propose you keep this but also make this a validation.